### DISCUSSION TITLE
ERC721 Function Overloading

### DISCUSSION THREAD

**Discussion by andrew-fleming** [OpenZeppelin/cairo-contracts#101] [Design]
# ERC721 Function Overloading

In order to exactly follow [EIP721](https://eips.ethereum.org/EIPS/eip-721), the proposal requires that two `safeTransferFrom()` functions exist; where, one includes a `bytes data` parameter and the other does not. Function overloading, as of now, is not possible with Cairo. Therefore, the current working implementation of ERC721 solely includes one `safeTransferFrom()` _with_ the extra parameter.

**Comment by martriay** [comment]
That solution seems reasonable to me. Let's make sure we document it.
